app/script/conversation/EventMapper.js

```diff
@@ -293,23 +293,14 @@ z.conversation.EventMapper = class EventMapper {
    */
   _mapEventMemberJoin(event, conversationEntity) {
     const {data: eventData, from: sender} = event;
-    const hasService = eventData.has_service;
-    let userIds = eventData.user_ids;
+    const {has_service: hasService, user_ids: userIds} = eventData;
 
     const messageEntity = new z.entity.MemberMessage();
 
-    const one2oneConversationTypes = [z.conversation.ConversationType.CONNECT, z.conversation.ConversationType.ONE2ONE];
     const messageFromCreator = sender === conversationEntity.creator;
-
-    if (one2oneConversationTypes.includes(conversationEntity.type())) {
-      const singleUserAdded = userIds.length === 1;
-      const isAcceptedConnection = messageFromCreator && singleUserAdded;
-      if (isAcceptedConnection) {
-        messageEntity.memberMessageType = z.message.SystemMessageType.CONNECTION_ACCEPTED;
-        userIds = conversationEntity.participating_user_ids();
-      } else {
-        messageEntity.visible(false);
-      }
+    const isSingleModeConversation = conversationEntity.is_one2one() || conversationEntity.is_request();
+    if (isSingleModeConversation) {
+      messageEntity.visible(false);
```

Comment:
What do you think of removing the `if` and doing `messageEntity.visible(!isSingleModeConversation)`?